addon/components/pie-chart.js

```diff
@@ -483,9 +483,17 @@ export default ChartComponent.extend(FloatingTooltipMixin,
     groups.select('path').attr(this.get('sliceAttrs'));
     
     var labelWidth = this.get('labelWidth');
+    var maxLabelWidth = this.get('outerWidth')/2 - this.get('labelPadding');
     var labelTrimmer = LabelTrimmer.create({
-      getLabelSize: function() {
-        return labelWidth;
+      getLabelSize: function(d, selection) {
+        var text, xPos;
+        // Pick the current text element by filtering on data
+        text = selection.filter(function(data) {
```

Comment:
Should I be using `select` instead? Both `filter` and `select` can create the same behavior but not sure why one should cause test failures. I can review this with you tomorrow.